### PR TITLE
DOC-853: Add warning about DROP COLUMN in multi statement transactions

### PR DESCRIPTION
**Pull request by nickvigilante**
Fixes DOC-853.

### CODE REVIEW

**Review by ajwerner — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12717)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @nickvigilante)

___
*[_includes/v21.2/known-limitations/schema-changes-within-transactions.md, line 9 at r1](https://reviewable.io/reviews/cockroachdb/docs/12717#-MtIyKF84_C5oY1fHsrI:-MtIyKF84_C5oY1fHsrJ:b-wq7ibt) ([raw file](https://github.com/cockroachdb/docs/blob/f5181edb2b4722e39edf19759ba5e872590f3312/_includes/v21.2/known-limitations/schema-changes-within-transactions.md#L9)):*
> ```Markdown
> - [Schema change DDL statements inside a multi-statement transaction can fail while other statements succeed](#schema-change-ddl-statements-inside-a-multi-statement-transaction-can-fail-while-other-statements-succeed).
> - As of v19.1, some schema changes can be used in combination in a single `ALTER TABLE` statement. For a list of commands that can be combined, see [`ALTER TABLE`](alter-table.html). For a demonstration, see [Add and rename columns atomically](rename-column.html#add-and-rename-columns-atomically).
> - `DROP COLUMN` can result in data loss if one of the other schema changes in the transaction fails or is canceled. To work around this, move the `DROP COLUMN` statement to its own transaction. See [tracking issue](https://github.com/cockroachdb/cockroach/issues/46541).
> ```

can you use language to make it clear that really what we mean is to run it in its own single-statement outside of a transaction or in its own explicit transaction. Given we want to run it alone, we're sorting telling folks not to use a transaction.

**Review by ajwerner — APPROVED** [review]
Do we want to add this note to older version too?

> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12717#-:-MtJE9ZL3yfqlO2mw3nb:b-8w9hjy)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @nickvigilante)

**Comment by nickvigilante** [comment]
@ajwerner Added for 20.2 and 21.1. Thanks!

File: v20.2/drop-column.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ toc: true
 
 The `DROP COLUMN` [statement](sql-statements.html) is part of `ALTER TABLE` and removes columns from a table.
 
+{{site.data.alerts.callout_danger}}
+When used in an explicit transaction combined with other schema changes to the same table, `DROP COLUMN` can result in data loss if one of the other schema change fails or is canceled. To work around this, move the `DROP COLUMN` statement to its own explicit transaction or run it in a single statement outside the existing transaction. See [tracking issue](https://github.com/cockroachdb/cockroach/issues/46541).
```
Comment (gemma-shay) [inline v20.2/drop-column.md:10]:
```suggestion
When used in an explicit transaction combined with other schema changes to the same table, `DROP COLUMN` can result in data loss if one of the other schema changes fails or is canceled. To work around this, move the `DROP COLUMN` statement to its own explicit transaction or run it in a single statement outside the existing transaction. See [tracking issue](https://github.com/cockroachdb/cockroach/issues/46541).
```

File: v20.2/drop-column.md [hunk 2]
```diff
@@ -6,6 +6,10 @@ toc: true
...
+{{site.data.alerts.end}}
+
 {{site.data.alerts.callout_info}}
 <span class="version-tag">New in v20.2:</span> By default, `DROP COLUMN` drops any indexes [indexes](indexes.html) on the column being dropped, and any indexes that reference the column, including [partial indexes](partial-indexes.html) with predicates that reference the column and indexes with [`STORING` clauses](create-index.html#store-columns) that reference the column.
```
Comment (gemma-shay) [inline v20.2/drop-column.md:14]:
```suggestion
<span class="version-tag">New in v20.2:</span> By default, `DROP COLUMN` drops any [indexes](indexes.html) on the column being dropped, and any indexes that reference the column, including [partial indexes](partial-indexes.html) with predicates that reference the column and indexes with [`STORING` clauses](create-index.html#store-columns) that reference the column.
```

File: v21.1/drop-column.md [hunk 3]
```diff
@@ -6,6 +6,10 @@ toc: true
 
 The `DROP COLUMN` [statement](sql-statements.html) is part of `ALTER TABLE` and removes columns from a table.
 
+{{site.data.alerts.callout_danger}}
+When used in an explicit transaction combined with other schema changes to the same table, `DROP COLUMN` can result in data loss if one of the other schema change fails or is canceled. To work around this, move the `DROP COLUMN` statement to its own explicit transaction or run it in a single statement outside the existing transaction. See [tracking issue](https://github.com/cockroachdb/cockroach/issues/46541).
```
Comment (gemma-shay) [inline v21.1/drop-column.md:10]: Same nits as 20.2

File: v21.2/drop-column.md [hunk 4]
```diff
@@ -6,6 +6,10 @@ toc: true
 
 The `DROP COLUMN` [statement](sql-statements.html) is part of `ALTER TABLE` and removes columns from a table.
 
+{{site.data.alerts.callout_danger}}
+When used in an explicit transaction combined with other schema changes to the same table, `DROP COLUMN` can result in data loss if one of the other schema change fails or is canceled. To work around this, move the `DROP COLUMN` statement to its own explicit transaction or run it in a single statement outside the existing transaction. See [tracking issue](https://github.com/cockroachdb/cockroach/issues/46541).
```
Comment (gemma-shay) [inline v21.2/drop-column.md:10]: Same nits as 20.2

**Review by gemma-shay — COMMENTED** [review]
Not sure if this applies here because this warning is about a limitation, but there was a [slack convo](https://cockroachlabs.slack.com/archives/C01NFCXULTX/p1632782818009300) a couple months ago about not using tracking issues outside of Known Limitations.

**Comment by nickvigilante** [comment]
@gemma-shay TFTR! That makes sense about excluding it from known limitations. That particular ticket is already referenced there anyway.

@lnhsingh Are you fine with me not including the tracking issue where I previously did?

**Review by lnhsingh — APPROVED** [review]
Yes - fine with you not including the track issue link.

Add a minor edit, applicable across all versions. Also, please add the Jira issue # to your PR description before you merge! Otherwise, LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12717#-:-MtPucYlAKEgoZoXAVNu:b-l9gfbs)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner, @gemma-shay, @lnhsingh, and @nickvigilante)

___
*[_includes/v20.2/known-limitations/schema-changes-within-transactions.md, line 9 at r5](https://reviewable.io/reviews/cockroachdb/docs/12717#-MtPuJcO1QyfrXG-N2MM:-MtPuJcO1QyfrXG-N2MN:byl4emd) ([raw file](https://github.com/cockroachdb/docs/blob/f065591e88e01d1150324a5151e93de78c5ced62/_includes/v20.2/known-limitations/schema-changes-within-transactions.md#L9)):*
> ```Markdown
> - [Schema change DDL statements inside a multi-statement transaction can fail while other statements succeed](#schema-change-ddl-statements-inside-a-multi-statement-transaction-can-fail-while-other-statements-succeed).
> - As of v19.1, some schema changes can be used in combination in a single `ALTER TABLE` statement. For a list of commands that can be combined, see [`ALTER TABLE`](alter-table.html). For a demonstration, see [Add and rename columns atomically](rename-column.html#add-and-rename-columns-atomically).
> - `DROP COLUMN` can result in data loss if one of the other schema changes in the transaction fails or is canceled. To work around this, move the `DROP COLUMN` statement to its own explicit transaction or run it in a single statement outside the existing transaction.
> ```

`DROP COLUMN` should link to drop-column.html
___
*[_includes/v21.1/known-limitations/schema-changes-within-transactions.md, line 9 at r5](https://reviewable.io/reviews/cockroachdb/docs/12717#-MtPuN8N9NJ0RKhDJrSH:-MtPuN8N9NJ0RKhDJrSI:b-ya81ng) ([raw file](https://github.com/cockroachdb/docs/blob/f065591e88e01d1150324a5151e93de78c5ced62/_includes/v21.1/known-limitations/schema-changes-within-transactions.md#L9)):*
> ```Markdown
> - [Schema change DDL statements inside a multi-statement transaction can fail while other statements succeed](#schema-change-ddl-statements-inside-a-multi-statement-transaction-can-fail-while-other-statements-succeed).
> - As of v19.1, some schema changes can be used in combination in a single `ALTER TABLE` statement. For a list of commands that can be combined, see [`ALTER TABLE`](alter-table.html). For a demonstration, see [Add and rename columns atomically](rename-column.html#add-and-rename-columns-atomically).
> - `DROP COLUMN` can result in data loss if one of the other schema changes in the transaction fails or is canceled. To work around this, move the `DROP COLUMN` statement to its own explicit transaction or run it in a single statement outside the existing transaction.
> ```

Same as above
___
*[_includes/v21.2/known-limitations/schema-changes-within-transactions.md, line 9 at r5](https://reviewable.io/reviews/cockroachdb/docs/12717#-MtPuOTpABowRujTEM2y:-MtPuOTpABowRujTEM2z:b-ya81ng) ([raw file](https://github.com/cockroachdb/docs/blob/f065591e88e01d1150324a5151e93de78c5ced62/_includes/v21.2/known-limitations/schema-changes-within-transactions.md#L9)):*
> ```Markdown
> - [Schema change DDL statements inside a multi-statement transaction can fail while other statements succeed](#schema-change-ddl-statements-inside-a-multi-statement-transaction-can-fail-while-other-statements-succeed).
> - As of v19.1, some schema changes can be used in combination in a single `ALTER TABLE` statement. For a list of commands that can be combined, see [`ALTER TABLE`](alter-table.html). For a demonstration, see [Add and rename columns atomically](rename-column.html#add-and-rename-columns-atomically).
> - `DROP COLUMN` can result in data loss if one of the other schema changes in the transaction fails or is canceled. To work around this, move the `DROP COLUMN` statement to its own explicit transaction or run it in a single statement outside the existing transaction.
> ```

Same as above

**Pull request merged**